src/create-glamorous.js

```diff
@@ -159,16 +176,54 @@ function createGlamorous(splitProps) {
       // set the displayName to something that's slightly more
       // helpful than `GlamorousComponent` :)
       displayName: displayName || `glamorous(${getDisplayName(comp)})`,
+      // these are props that should be applied to the component at render time
+      propsToApply,
     }
   }
+}
 
-  function when(comp, prop) {
-    return comp ? comp.concat(prop) : prop
-  }
+/**
+ * reduces the propsToApply given to a single props object
+ * @param {Array} propsToApply an array of propsToApply objects:
+ *   - object
+ *   - array of propsToApply items
+ *   - function that accepts the accumulated props and the context
+ * @param {Object} accumulator an object to apply props onto
+ * @param {Object} props the props that should ultimately take precedence
+ * @param {*} context the context object
+ * @return {Object} the reduced props
+ */
+function getPropsToApply(propsToApply, accumulator, props, context) {
+  // using forEach rather than reduce here because the reduce solution
+  // effectively did the same thing because we manipulate the `accumulator`
```

Comment:
Yeah, i guess that does make sense. Each iteration in the reducer would just carry over the same object reference. Makes sense.